src/main/scala/com/fulcrumgenomics/umi/CorrectUmis.scala

```
@@ -260,6 +262,15 @@ class CorrectUmis
       if (wrongLengthRecords > 0) logger.error(s"# ${wrongLengthRecords} had unexpected UMIs of differing lengths in the BAM file!")
       logger.error("###################################################################")
     }
+
+    minCorrected.foreach{ min =>
+      val ratioKept = 1.0 * kept / totalRecords
+      if (ratioKept < min) {
+        logger.error("###################################################################")
+        logger.error(f"# Final ratio of reads kept / total was ${ratioKept}%2.2f (user specified minimum was ${min}%2.2f)")
+        logger.error("###################################################################")
```

did you want to throw an exception?